### ISSUE TITLE
Header merge issue in Webmock 2.x

### ISSUE THREAD

**Issue by grosscr** [bblimke/webmock#620]
So I was working on upgrading one of our repos to webmock 2.0.2 and addressing the changes to the basic auth stubbing. In the process, I believe I found an issue in webmock related to the use of handsoap w/ the curb driver. Curl::Easy (used by the curb driver) allows for request headers to be set and sent in the form of an array instead of a hash ([see here](http://www.rubydoc.info/github/taf2/curb/Curl/Easy#headers%3D-instance_method)). When the request headers are an array that makes the [basic_auth_headers merge](https://github.com/bblimke/webmock/blob/bf695d1dc9c541bfee69a88634b47286574e80e1/lib/webmock/http_lib_adapters/curb_adapter.rb#L101) fail, because you can't merge into an array.

Partial stack trace:
```
       expected no Exception, got #<NoMethodError: undefined method `merge' for #<Array:0x007fe36cefa338>> with backtrace:
         # /Library/Ruby/Gems/2.0.0/gems/webmock-2.0.2/lib/webmock/http_lib_adapters/curb_adapter.rb:101:in `build_request_signature'
         # /Library/Ruby/Gems/2.0.0/gems/webmock-2.0.2/lib/webmock/http_lib_adapters/curb_adapter.rb:59:in `curb_or_webmock'
         # /Library/Ruby/Gems/2.0.0/gems/webmock-2.0.2/lib/webmock/http_lib_adapters/curb_adapter.rb:222:in `perform'
         # /Library/Ruby/Gems/2.0.0/gems/webmock-2.0.2/lib/webmock/http_lib_adapters/curb_adapter.rb:216:in `http_post'
         # /Library/Ruby/Gems/2.0.0/gems/webmock-2.0.2/lib/webmock/http_lib_adapters/curb_adapter.rb:216:in `http_post'
         # /Library/Ruby/Gems/2.0.0/gems/handsoap-1.4.0/lib/handsoap/http/drivers/curb_driver.rb:64:in `send_http_request'
         # /Library/Ruby/Gems/2.0.0/gems/handsoap-1.4.0/lib/handsoap/service.rb:262:in `invoke'
```

And we were able to reproduce it with the following test addition to your specs:

```
diff --git a/spec/acceptance/curb/curb_spec.rb b/spec/acceptance/curb/curb_spec.rb
index d939874..67b0e21 100644
--- a/spec/acceptance/curb/curb_spec.rb
+++ b/spec/acceptance/curb/curb_spec.rb
@@ -355,6 +355,16 @@ unless RUBY_PLATFORM =~ /java/
         c.http(:GET)
         expect(c.body).to eq("abc")
       end
+
+      it "supports array headers passed to Curl::Easy" do
+        stub_request(:get, "www.example.com").to_return(:body => "abc")
+
+        c = Curl::Easy.new
+        c.url = "http://www.example.com"
+        c.headers = ["X-One: 1"]
+        c.http(:GET)
+        expect(c.body).to eq("abc")
+      end
     end
 
     describe "using #http_* methods for requests" do

Failures:

  1) Webmock with Curb using #http for requests supports array headers passed to Curl::Easy
     Failure/Error: :headers => self.headers.merge(basic_auth_headers)

     NoMethodError:
       undefined method `merge' for ["X-One: 1"]:Array
     # ./lib/webmock/http_lib_adapters/curb_adapter.rb:101:in `build_request_signature'
     # ./lib/webmock/http_lib_adapters/curb_adapter.rb:59:in `curb_or_webmock'
     # ./lib/webmock/http_lib_adapters/curb_adapter.rb:222:in `perform'
     # ./lib/webmock/http_lib_adapters/curb_adapter.rb:196:in `http'
     # ./lib/webmock/http_lib_adapters/curb_adapter.rb:196:in `http'
     # ./spec/acceptance/curb/curb_spec.rb:365:in `block (3 levels) in <top (required)>'
```

**Comment by bblimke** [comment]
Fix released in version 2.0.3

**Comment by grosscr** [comment]
Thanks for the quick turn around!
